buku

```diff
@@ -1628,79 +1680,77 @@ class BukuDb:
             delimited with ','.
             Retrieves entries matching ALL tags if tags are
             delimited with '+'.
+        order : list of str
+            Order description (fields from JSON export or DB, prepended with '+'/'-' for ASC/DESC).
+            Note: this applies to fields with the same number of matched tags.
 
         Returns
         -------
         list
             List of search results.
         """
 
+        _order = self._order(order)
         LOGDBG(tags)
         if tags is None or tags == DELIM or tags == '':
             return []
 
-        tags_, search_operator, excluded_tags = prep_tag_search(tags)
+        qargs, search_operator, excluded_tags = prep_tag_search(tags)
         if search_operator is None:
             LOGERR("Cannot use both '+' and ',' in same search")
             return []
 
-        LOGDBG('tags: %s', tags_)
+        LOGDBG('tags: %s', qargs)
         LOGDBG('search_operator: %s', search_operator)
         LOGDBG('excluded_tags: %s', excluded_tags)
 
         if search_operator == 'AND':
-            query = ("SELECT id, url, metadata, tags, desc, flags FROM bookmarks "
-                     "WHERE tags LIKE '%' || ? || '%' ")
-            for tag in tags_[1:]:
-                query += "{} tags LIKE '%' || ? || '%' ".format(search_operator)
-
-            if excluded_tags:
-                tags_.append(excluded_tags)
-                query = query.replace('WHERE tags', 'WHERE (tags')
-                query += ') AND tags NOT REGEXP ? '
-            query += 'ORDER BY id ASC'
+            query = ('SELECT id, url, metadata, tags, desc, flags FROM bookmarks WHERE (' +
+                     f' {search_operator} '.join("tags LIKE '%' || ? || '%'" for tag in qargs) +
+                     ')' + ('' if not excluded_tags else ' AND tags NOT REGEXP ?') +
+                     f' ORDER BY {_order}')
```

Comment:
Other than replacing the `ORDER BY` clause, the changes here don't really affect the produced SQL.